cmake/kokkos_arch.cmake

```
@@ -92,7 +92,12 @@ ENDIF()
 IF(KOKKOS_ENABLE_COMPILER_WARNINGS)
   SET(COMMON_WARNINGS
     "-Wall" "-Wunused-parameter" "-Wshadow" "-pedantic"
-    "-Wsign-compare" "-Wtype-limits" "-Wuninitialized")
+    "-Wtype-limits" "-Wuninitialized")
+
+  # OpenMPTarget compilers give erroneous warnings about sign comparison in loops
+  IF(NOT KOKKOS_ENABLE_OPENMPTARGET)
+    LIST(APPEND COMMON_WARNINGS "-Wsign-compare")
+  ENDIF()
```

Change to
```suggestion
    "-Wsign-compare" "-Wtype-limits" "-Wuninitialized")

  # OpenMPTarget compilers give erroneous warnings about sign comparison in loops
  IF(KOKKOS_ENABLE_OPENMPTARGET)
    LIST(REMOVE_ITEM COMMON_WARNINGS "-Wsign-compare")
  ENDIF()
```
so it is clear this is temporary workaround for OpenMP target and it does not hurt readability for the most common case where one want to see what warnings are enabled.